src/relax/op/tensor/binary.cc

```diff
@@ -100,10 +114,69 @@ StructInfo InferStructInfoBroadcast(const Call& call, const BlockBuilder& ctx) {
     }
     Expr shape = ShapeExpr(Array<PrimExpr>(output_shape.rbegin(), output_shape.rend()));
     return TensorStructInfo(shape, output_dtype);
+  } else if (lhs_shape != nullptr && lhs_sinfo->ndim == output_ndim) {
+    for (int i = 0; i < output_ndim; ++i) {
+      if (!IsIntGreaterThanOne(lhs_shape->values[i])) {
```

Comment:
While it is OK to be smart in this case and do this kind of deduction. Let us keep things simple, and simply do the fallback logic here.

Since there are indeed extra proofs needed to ensure rhs can broadcast with lhs. This is also not a common case

Reply:
So we can delete the two else cases for code simplicity

Reply:
This part tries to cover the cases like `lhs = Tensor((2, 3))` and `rhs = Tensor(ndim=2)`. In such cases, either it runs into runtime error (shape not compatible), or the output shape will be `(2, 3)` otherwise. The runtime part is actually not provable and that was why I added this.

Reply:
I understand the goals here. However, like mentioned above I still think it is a less common case and it is fine to return tensor ndim dtype instead for simplicity